### PR TITLE
Fixes #25298 - Write audit log on errata installation

### PR DESCRIPTION
**Pull request by sbernhard**

### CODE REVIEW

**Comment by sbernhard** [comment]
I really don't know why this test doesn't work. Would be nice if someone can help.

**Comment by beav** [comment]
[test katello]

It looks like the test got SIGTERM last time

**Comment by beav** [comment]
looks like this test failure is related to the PR, but probably is a small fix

```
Actions::Katello::Host::Erratum::InstallTest#test_0002_plans installs with batching = 0.74 s = F


Failure:
Actions::Katello::Host::Erratum::InstallTest#test_0002_plans installs with batching [/usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-3745/gems/dynflow-1.1.2/lib/dynflow/middleware.rb:35]:
unexpected invocation: #<Mock:a_host>.id()
unsatisfied expectations:
- expected exactly once, invoked twice: #<Mock:a_host>.id(any_parameters)
satisfied expectations:
- allowed any number of times, not yet invoked: Katello::Ping.ping(any_parameters)
- allowed any number of times, not yet invoked: Cert::Certs.ssl_client_key(any_parameters)
- allowed any number of times, not yet invoked: Cert::Certs.ssl_client_cert(any_parameters)
- allowed any number of times, not yet invoked: Cert::Certs.ca_cert(any_parameters)
- allowed any number of times, invoked 6 times: Setting(id: integer, name: string, value: text, description: text, category: string, settings_type: string, default: text, created_at: datetime, updated_at: datetime, full_name: string, encrypted: boolean).[](any_parameters)
- allowed any number of times, invoked once: #<Actions::Katello::Host::Erratum::Install:0x14413138>.action_subject(#<Mock:a_host>, {:errata => ["RHBA-2014-1234", "RHBA-2014-1235", "RHBA-2014-1236", "RHBA-2014-1237"]})
- allowed any number of times, invoked once: #<Actions::Katello::Host::Erratum::Install:0x14413138>.action_subject(#<Mock:a_host>, {:errata => ["RHBA-2014-1234"]})
- expected exactly once, invoked once: #<Actions::Katello::Host::Erratum::Install:0x14413138>.plan_self(any_parameters)
- allowed any number of times, invoked twice: #<Mock:a_system>.uuid(any_parameters)
- expected exactly once, invoked once: #<Mock:a_system>.uuid(any_parameters)
- allowed any number of times, invoked twice: #<Mock:a_host>.content_facet(any_parameters)
- expected exactly once, invoked once: #<Mock:a_host>.content_facet(any_parameters)


bin/rails test /var/lib/workspace/workspace/katello-pr-test/test/actions/katello/host/erratum_test.rb:56
```

**Comment by sbernhard** [comment]
Thanks @beav. Can you give me a hint? Don't know whats wrong with the code...

**Comment by sbernhard** [comment]
done. :-)

**Comment by beav** [comment]
thx, I tried it last night but was not able to figure out where it was breaking :cat:

File: app/lib/actions/katello/host/erratum/install.rb [hunk 1]
```diff
@@ -39,6 +40,11 @@ def resource_locks
           def presenter
             Helpers::Presenter::Delegated.new(self, planned_actions(Pulp::Consumer::ContentInstall))
           end
+
+          def finalize
+            host = ::Host.find_by(:id => input[:host_id])
+            host.update(audit_comment: _("Installation of errata requested: %{errata}") % {errata: input[:errata].join(", ")})
```
Comment (beav) [inline app/lib/actions/katello/host/erratum/install.rb]: where is `input[:errata]` being defined? The `plan_self` only has `:host_id` in it

Comment (sbernhard) [reply app/lib/actions/katello/host/erratum/install.rb]: I guess, the input[:errata] is send the same way as it is send in humanized_input. 
It works. 

image

**Comment by sbernhard** [comment]
@beav, its now ready to test.

**Comment by beav** [comment]
@sbernhard I am a little behind on items but will take a look soon, sorry for the delay.

**Comment by beav** [comment]
thanks @sbernhard !

**Pull request merged**